draft-nygren-dnsop-svcb-httpssvc.md

```diff
@@ -627,6 +627,20 @@ in cache before performing any followup queries.
 With these optimizations in place, and conforming DNS servers,
 using SVCB does not add network latency to connection setup.
 
+Moreover, if an A or AAAA response arrives before an SVCB response, a client which 
+prefers ESNI SHOULD wait up to CD milliseconds before starting secure connections 
```

Comment:
The current logic just requires the client to wait for the SVCB response.  As you note, the SVCB response contains security-relevant information, so it's important not to miss it.

Is the goal here to handle [DNS non-response](https://tools.ietf.org/html/draft-ietf-dnsop-no-response-issue-13)?  Even the most ancient DNS servers will return an error if there is a problem with the SVCB query.  Total non-response is a serious bug, but it does happen.

As that draft notes, the client [can measure this behavior](https://tools.ietf.org/html/draft-ietf-dnsop-no-response-issue-13#section-3.1.2).  If you think it's worth documenting a solution for this case, I'd prefer that approach, and I'd suggest doing it in a separate "Compatibility" section.

Reply:
> Is the goal here to handle DNS non-response?

Not quite! The goal is to handle answers received out of order. It’s an optimization.

Reply:
OK.  As your text notes, SVCB covers a variety of transports and use cases, so I think we need to express this with more generality, like:
> If an A or AAAA response arrives before an SVCB response, the client MAY begin connection as if the SVCB query returned NODATA, but MUST NOT transmit any information that could be altered by an SVCB response until the SVCB response arrives. For example, ...(esni)... .

And then separately
> If an SVCB RR is consistent with an active or in-progress connection, the client MAY prefer that RR.

Reply:
Excellent suggestion! I tweaked your proposed text and added it. Please have a look!